From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Biquaternions <biquaternions@serlith.net>
Date: Mon, 29 Jun 2026 21:10:17 -0500
Subject: [PATCH] Make Item Cooldown Toggleable


diff --git a/net/minecraft/server/level/ServerPlayer.java b/net/minecraft/server/level/ServerPlayer.java
index 3cd4e850f223100b61ac6831a87dd38f3a3bb58f..8c68b180b0871a5a068025e07d5ac918041a4859 100644
--- a/net/minecraft/server/level/ServerPlayer.java
+++ b/net/minecraft/server/level/ServerPlayer.java
@@ -712,7 +712,7 @@ public class ServerPlayer extends Player implements ca.spottedleaf.moonrise.patc
 
     @Override
     protected ItemCooldowns createItemCooldowns() {
-        return new ServerItemCooldowns(this);
+        return new me.biquaternions.nitrogen.world.item.NoopItemCooldowns(); // Nitrogen - Disable cooldown
     }
 
     @Override
