From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Biquaternions <biquaternions@serlith.net>
Date: Tue, 18 Aug 2026 22:13:35 -0500
Subject: [PATCH] Configurable Combat Profiles


diff --git a/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index 9a848431f86fccc284b3090a8285dd19a263b15e..6e5fa810004b32aefc9a5e6c2254d2d5de638d4f 100644
--- a/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -1106,7 +1106,7 @@ public class ServerGamePacketListenerImpl
 
             this.send(new ClientboundSetHeldSlotPacket(inventory.getSelectedSlot()));
             this.player.inventoryMenu.broadcastChanges();
-            if (io.papermc.paper.configuration.GlobalConfiguration.get().unsupportedSettings.updateEquipmentOnPlayerActions) this.player.detectEquipmentUpdates(); // Paper - Force update attributes.
+            if (!this.player.getBukkitEntity().getCombatProfile().canAttributeSwap()) this.player.detectEquipmentUpdates(); // Paper - Force update attributes. // Nitrogen - Combat Profiles
         }
     }
 
@@ -1982,7 +1982,7 @@ public class ServerGamePacketListenerImpl
                         }
                         // CraftBukkit end
                         this.player.stopUsingItem();
-                        if (io.papermc.paper.configuration.GlobalConfiguration.get().unsupportedSettings.updateEquipmentOnPlayerActions) this.player.detectEquipmentUpdates(); // Paper - Force update attributes.
+                        if (!this.player.getBukkitEntity().getCombatProfile().canAttributeSwap()) this.player.detectEquipmentUpdates(); // Paper - Force update attributes. // Nitrogen - Combat Profiles
                     }
 
                     return;
@@ -2004,20 +2004,20 @@ public class ServerGamePacketListenerImpl
                         }
                         // CraftBukkit end
                         this.player.drop(false);
-                        if (io.papermc.paper.configuration.GlobalConfiguration.get().unsupportedSettings.updateEquipmentOnPlayerActions) this.player.detectEquipmentUpdates(); // Paper - Force update attributes.
+                        if (!this.player.getBukkitEntity().getCombatProfile().canAttributeSwap()) this.player.detectEquipmentUpdates(); // Paper - Force update attributes. // Nitrogen - Combat Profiles
                     }
 
                     return;
                 case DROP_ALL_ITEMS:
                     if (!this.player.isSpectator()) {
                         this.player.drop(true);
-                        if (io.papermc.paper.configuration.GlobalConfiguration.get().unsupportedSettings.updateEquipmentOnPlayerActions) this.player.detectEquipmentUpdates(); // Paper - Force update attributes.
+                        if (!this.player.getBukkitEntity().getCombatProfile().canAttributeSwap()) this.player.detectEquipmentUpdates(); // Paper - Force update attributes. // Nitrogen - Combat Profiles
                     }
 
                     return;
                 case RELEASE_USE_ITEM:
                     this.player.releaseUsingItem();
-                    if (io.papermc.paper.configuration.GlobalConfiguration.get().unsupportedSettings.updateEquipmentOnPlayerActions) this.player.detectEquipmentUpdates(); // Paper - Force update attributes.
+                    if (!this.player.getBukkitEntity().getCombatProfile().canAttributeSwap()) this.player.detectEquipmentUpdates(); // Paper - Force update attributes. // Nitrogen - Combat Profiles
                     return;
                 case START_DESTROY_BLOCK:
                 case ABORT_DESTROY_BLOCK:
@@ -2050,7 +2050,7 @@ public class ServerGamePacketListenerImpl
                         }
                     }
                     // Paper end - Send block entities after destroy prediction
-                    if (io.papermc.paper.configuration.GlobalConfiguration.get().unsupportedSettings.updateEquipmentOnPlayerActions) this.player.detectEquipmentUpdates(); // Paper - Force update attributes.
+                    if (!this.player.getBukkitEntity().getCombatProfile().canAttributeSwap()) this.player.detectEquipmentUpdates(); // Paper - Force update attributes. // Nitrogen - Combat Profiles
                     return;
                 default:
                     throw new IllegalArgumentException("Invalid player action");
@@ -2173,7 +2173,7 @@ public class ServerGamePacketListenerImpl
                             this.send(new ClientboundBlockUpdatePacket(level, pos));
                             this.send(new ClientboundBlockUpdatePacket(level, pos.relative(direction)));
                         }
-                        if (io.papermc.paper.configuration.GlobalConfiguration.get().unsupportedSettings.updateEquipmentOnPlayerActions) this.player.detectEquipmentUpdates(); // Paper - Force update attributes.
+                        if (!this.player.getBukkitEntity().getCombatProfile().canAttributeSwap()) this.player.detectEquipmentUpdates(); // Paper - Force update attributes. // Nitrogen - Combat Profiles
                         // Paper - Remove unused warning
                     }
                 }
@@ -2332,7 +2332,7 @@ public class ServerGamePacketListenerImpl
 
             this.player.getInventory().setSelectedSlot(packet.getSlot());
             this.player.resetLastActionTime();
-            if (io.papermc.paper.configuration.GlobalConfiguration.get().unsupportedSettings.updateEquipmentOnPlayerActions) this.player.detectEquipmentUpdates(); // Paper - Force update attributes.
+            if (!this.player.getBukkitEntity().getCombatProfile().canAttributeSwap()) this.player.detectEquipmentUpdates(); // Paper - Force update attributes. // Nitrogen - Combat Profiles
         } else {
             LOGGER.warn("{} tried to set an invalid carried item", this.player.getPlainTextName());
             this.disconnect(Component.literal("Invalid hotbar selection (Hacking?)"), org.bukkit.event.player.PlayerKickEvent.Cause.ILLEGAL_ACTION); // CraftBukkit // Paper - kick event cause
@@ -2870,7 +2870,7 @@ public class ServerGamePacketListenerImpl
                 CraftEventFactory.callPlayerUseUnknownEntityEvent(ServerGamePacketListenerImpl.this.player, packet.entityId(), true, net.minecraft.world.InteractionHand.MAIN_HAND, null);
             }
             // Paper end - PlayerUseUnknownEntityEvent
-            if (io.papermc.paper.configuration.GlobalConfiguration.get().unsupportedSettings.updateEquipmentOnPlayerActions) this.player.detectEquipmentUpdates(); // Paper - Force update attributes.
+            if (!this.player.getBukkitEntity().getCombatProfile().canAttributeSwap()) this.player.detectEquipmentUpdates(); // Paper - Force update attributes. // Nitrogen - Combat Profiles
         }
     }
 
@@ -2962,7 +2962,7 @@ public class ServerGamePacketListenerImpl
                 CraftEventFactory.callPlayerUseUnknownEntityEvent(ServerGamePacketListenerImpl.this.player, packet.entityId(), false, packet.hand(), packet.location());
             }
             // Paper end - PlayerUseUnknownEntityEvent
-            if (io.papermc.paper.configuration.GlobalConfiguration.get().unsupportedSettings.updateEquipmentOnPlayerActions) this.player.detectEquipmentUpdates(); // Paper - Force update attributes.
+            if (!this.player.getBukkitEntity().getCombatProfile().canAttributeSwap()) this.player.detectEquipmentUpdates(); // Paper - Force update attributes. // Nitrogen - Combat Profiles
         }
     }
 
@@ -3377,7 +3377,7 @@ public class ServerGamePacketListenerImpl
                         this.player.containerMenu.broadcastChanges();
                     }
                     if (packet.buttonNum() == Inventory.SLOT_OFFHAND && this.player.containerMenu != this.player.inventoryMenu) this.player.containerSynchronizer.sendOffHandSlotChange(); // Paper - update offhand data when the player is clicking in an inventory not their own as the sychronizer does not include offhand slots
-                    if (io.papermc.paper.configuration.GlobalConfiguration.get().unsupportedSettings.updateEquipmentOnPlayerActions) this.player.detectEquipmentUpdates(); // Paper - Force update attributes.
+                    if (!this.player.getBukkitEntity().getCombatProfile().canAttributeSwap()) this.player.detectEquipmentUpdates(); // Paper - Force update attributes. // Nitrogen - Combat Profiles
                 }
             }
         }
@@ -3470,7 +3470,7 @@ public class ServerGamePacketListenerImpl
                 if (clickAccepted) {
                     this.player.containerMenu.broadcastChanges();
                 }
-                if (io.papermc.paper.configuration.GlobalConfiguration.get().unsupportedSettings.updateEquipmentOnPlayerActions) this.player.detectEquipmentUpdates(); // Paper - Force update attributes.
+                if (!this.player.getBukkitEntity().getCombatProfile().canAttributeSwap()) this.player.detectEquipmentUpdates(); // Paper - Force update attributes. // Nitrogen - Combat Profiles
             }
         }
     }
@@ -3528,7 +3528,7 @@ public class ServerGamePacketListenerImpl
                 this.player.inventoryMenu.getSlot(packet.slotNum()).setByPlayer(itemStack);
                 this.player.inventoryMenu.setRemoteSlot(packet.slotNum(), itemStack);
                 this.player.inventoryMenu.broadcastChanges();
-                if (io.papermc.paper.configuration.GlobalConfiguration.get().unsupportedSettings.updateEquipmentOnPlayerActions) this.player.detectEquipmentUpdates(); // Paper - Force update attributes.
+                if (!this.player.getBukkitEntity().getCombatProfile().canAttributeSwap()) this.player.detectEquipmentUpdates(); // Paper - Force update attributes. // Nitrogen - Combat Profiles
             } else if (drop && validData) {
                 if (this.dropSpamThrottler.isUnderThreshold()) {
                     this.dropSpamThrottler.increment();
diff --git a/net/minecraft/world/entity/LivingEntity.java b/net/minecraft/world/entity/LivingEntity.java
index 398e248f8916894030172d7e50ed5db832b4a03f..6ca7bd1bac5ee204b7bdb65277c7cf9a1e7fa70f 100644
--- a/net/minecraft/world/entity/LivingEntity.java
+++ b/net/minecraft/world/entity/LivingEntity.java
@@ -2067,6 +2067,12 @@ public abstract class LivingEntity extends Entity implements Attackable, Waypoin
     }
 
     public void knockback(double power, double xd, double zd, final DamageSource source, final float damage, final boolean comesFromEffect, @Nullable Entity attacker, io.papermc.paper.event.entity.EntityKnockbackEvent.Cause eventCause) { // Paper - knockback events
+        // Nitrogen start - Configurable Legacy Knockback
+        final org.bukkit.craftbukkit.damage.CraftDamageSource craftDamageSource = new org.bukkit.craftbukkit.damage.CraftDamageSource(source);
+        final me.biquaternions.nitrogen.entity.knockback.KnockbackBehavior behavior = this.getBukkitLivingEntity().getCombatProfile().getKnockback().getBehaviorFromDamageSource(craftDamageSource);
+        behavior.handleKnockback(this.getBukkitLivingEntity(), power, xd, zd, craftDamageSource, damage, comesFromEffect, attacker.getBukkitEntity(), eventCause);
+        if (true) return;
+        // Nitrogen end - Configurable Legacy Knockback
         power *= 1.0 - this.getAttributeValue(Attributes.KNOCKBACK_RESISTANCE);
         if (true || !(power <= 0.0)) { // CraftBukkit - Call event even when force is 0
             // this.needsSync = true; // Paper - moved down
diff --git a/net/minecraft/world/entity/ai/attributes/Attributes.java b/net/minecraft/world/entity/ai/attributes/Attributes.java
index 54a8384f1bc805865e81aeebcf1eadc89d097c91..94fac0a450e286ed92396bc1af650d9b126ac0af 100644
--- a/net/minecraft/world/entity/ai/attributes/Attributes.java
+++ b/net/minecraft/world/entity/ai/attributes/Attributes.java
@@ -59,6 +59,7 @@ public class Attributes {
         "jump_strength", new RangedAttribute("attribute.name.jump_strength", 0.42F, 0.0, 32.0).setSyncable(true)
     );
     public static final Holder<Attribute> KNOCKBACK_RESISTANCE = register(
+        // Nitrogen - Can't change 'minValue' without breaking something
         "knockback_resistance", new RangedAttribute("attribute.name.knockback_resistance", 0.0, -2.0, 1.0)
     );
     public static final Holder<Attribute> LUCK = register("luck", new RangedAttribute("attribute.name.luck", 0.0, -1024.0, 1024.0).setSyncable(true));
diff --git a/net/minecraft/world/entity/player/Player.java b/net/minecraft/world/entity/player/Player.java
index 2a71a2fa949b3a7fb8ff2c44ec3a3d8636237142..954f4ac1e5857a0e8d7a56d851d8c938e265abaf 100644
--- a/net/minecraft/world/entity/player/Player.java
+++ b/net/minecraft/world/entity/player/Player.java
@@ -1053,7 +1053,7 @@ public abstract class Player extends Avatar implements ContainerUser {
                     criticalAttack = criticalAttack && !this.level().paperConfig().entities.behavior.disablePlayerCrits; // Paper - Toggleable player crits
                     if (criticalAttack) {
                         damageSource = damageSource.critical(); // Paper - critical damage API
-                        baseDamage *= 1.5F;
+                        baseDamage *= this.getBukkitEntity().getCombatProfile().getCriticalMultiplier(); // Nitrogen - Combat Profiles
                     }
 
                     float totalDamage = baseDamage + magicBoost;
@@ -1066,8 +1066,20 @@ public abstract class Player extends Avatar implements ContainerUser {
                     Vec3 oldMovement = entity.getDeltaMovement();
                     boolean wasHurt = entity.hurtOrSimulate(damageSource, totalDamage);
                     if (wasHurt) {
+                        // Nitrogen start - Extra knockback multiplier
+                        float knockbackMultiplier = 0.0F;
+                        if (knockbackAttack) {
+                            // Players get an extra knockback from the enchantment level + 1 if they're sprinting
+                            // In modern we use 'knockbackAttack' which is only true when is sprinting and a full strength attack
+                            // Nitrogen will disable the cooldown logic, so all attacks will be on full strength
+                            // See: https://papersrc.biquaternions.me/1/paper-1.8.8/net/minecraft/server/EntityHuman#L944
+                            final org.bukkit.craftbukkit.damage.CraftDamageSource craftDamageSource = new org.bukkit.craftbukkit.damage.CraftDamageSource(damageSource);
+                            final me.biquaternions.nitrogen.entity.knockback.KnockbackBehavior behavior = this.getBukkitEntity().getCombatProfile().getKnockback().getBehaviorFromDamageSource(craftDamageSource);
+                            knockbackMultiplier = behavior.getMultiplierExtra();
+                        }
+                        // Nitrogen end - Extra knockback multiplier
                         this.causeExtraKnockback(
-                            entity, this.getKnockback(entity, damageSource) + (knockbackAttack ? 0.5F : 0.0F), oldMovement, damageSource, totalDamage, true
+                            entity, this.getKnockback(entity, damageSource) + knockbackMultiplier, oldMovement, damageSource, totalDamage, true // Nitrogen - Extra knockback multiplier
                         );
                         if (sweepAttack) {
                             this.doSweepAttack(entity, baseDamage, damageSource, attackStrengthScale);
@@ -1211,6 +1223,16 @@ public abstract class Player extends Avatar implements ContainerUser {
         final boolean comesFromEffect
     ) {
         if (knockbackAmount > 0.0F) {
+            // Nitrogen start - Handle Knockback extra
+            final org.bukkit.craftbukkit.damage.CraftDamageSource craftDamageSource = new org.bukkit.craftbukkit.damage.CraftDamageSource(damageSource);
+            final me.biquaternions.nitrogen.entity.CombatProfile profile = this.getBukkitEntity().getCombatProfile();
+            final me.biquaternions.nitrogen.entity.knockback.KnockbackBehavior behavior = profile.getKnockback().getBehaviorFromDamageSource(craftDamageSource);
+            behavior.handleExtraKnockback(this.getBukkitEntity(), entity.getBukkitEntity(), knockbackAmount, org.bukkit.craftbukkit.util.CraftVector.toBukkit(oldMovement), craftDamageSource, damage, comesFromEffect);
+            if (profile.interruptsSprintOnHit()) {
+                this.setSprinting(false);
+            }
+            if (false) {
+            // Nitrogen end - Handle Knockback extra
             if (entity instanceof LivingEntity livingTarget) {
                 livingTarget.knockback(
                     knockbackAmount, Mth.sin(this.getYRot() * Mth.DEG_TO_RAD), -Mth.cos(this.getYRot() * Mth.DEG_TO_RAD), damageSource, damage, comesFromEffect, this, io.papermc.paper.event.entity.EntityKnockbackEvent.Cause.ENTITY_ATTACK // Paper - knockback events
@@ -1225,6 +1247,7 @@ public abstract class Player extends Avatar implements ContainerUser {
             this.setSprinting(false);
             }
             // Paper end - Configurable sprint interruption on attack
+            } // Nitrogen - Cancel sprint on hit
         }
 
         if (entity instanceof ServerPlayer serverPlayer && entity.hurtMarked) {
@@ -1993,7 +2016,7 @@ public abstract class Player extends Avatar implements ContainerUser {
     public void detectEquipmentUpdates() {
         super.detectEquipmentUpdates();
 
-        if (!io.papermc.paper.configuration.GlobalConfiguration.get().unsupportedSettings.updateEquipmentOnPlayerActions) {
+        if (this.getBukkitEntity().getCombatProfile().canAttributeSwap()) { // Nitrogen - Combat Profiles
             return;
         }
 
diff --git a/net/minecraft/world/entity/projectile/Projectile.java b/net/minecraft/world/entity/projectile/Projectile.java
index bbb088d7e94d4ad808f8d20a449a2f377934666c..eb4c33bc5ffd4190f44abf0c77d5f9a37b290cc6 100644
--- a/net/minecraft/world/entity/projectile/Projectile.java
+++ b/net/minecraft/world/entity/projectile/Projectile.java
@@ -191,7 +191,7 @@ public abstract class Projectile extends Entity implements TraceableEntity {
         if (!sourceMovement.isFinite()) {
             sourceMovement = Vec3.ZERO;
         }
-        if (!source.level().paperConfig().misc.disableRelativeProjectileVelocity) {
+        if (!(source instanceof LivingEntity living) || living.getBukkitLivingEntity().getCombatProfile().projectilesInheritVelocity()) { // Nitrogen - Self hit by projectiles
         this.setDeltaMovement(this.getDeltaMovement().add(sourceMovement.x, source.onGround() ? 0.0 : sourceMovement.y, sourceMovement.z));
         }
         // Paper end - allow disabling relative velocity
