From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Biquaternions <biquaternions@serlith.net>
Date: Wed, 12 Aug 2026 17:16:39 -0500
Subject: [PATCH] Disable Dynamic Attack Strength


diff --git a/net/minecraft/world/entity/player/Player.java b/net/minecraft/world/entity/player/Player.java
index c9f61291866bb0cf7ebb7aac698a233de191103c..211e842f455c45ec380568c98926443f2dc8aac8 100644
--- a/net/minecraft/world/entity/player/Player.java
+++ b/net/minecraft/world/entity/player/Player.java
@@ -1136,6 +1136,7 @@ public abstract class Player extends Avatar implements ContainerUser {
     }
 
     private boolean isSweepAttack(final boolean fullStrengthAttack, final boolean criticalAttack, final boolean knockbackAttack) {
+        if (true) return false; // Nitrogen - Disable sweep attack
         if (fullStrengthAttack && !criticalAttack && !knockbackAttack && this.onGround()) {
             double approximateSpeedSq = this.getKnownMovement().horizontalDistanceSqr();
             double maxSpeedForSweepAttack = this.getSpeed() * 2.5;
@@ -1975,6 +1976,7 @@ public abstract class Player extends Avatar implements ContainerUser {
     }
 
     public float getCurrentItemAttackStrengthDelay() {
+        if (true) return 0.0F; // Nitrogen - Disable dynamic attack strength - API has to reflect no delay
         return (float)(1.0 / this.getAttributeValue(Attributes.ATTACK_SPEED) * 20.0);
     }
 
@@ -1985,6 +1987,7 @@ public abstract class Player extends Avatar implements ContainerUser {
     }
 
     public float getAttackStrengthScale(final float a) {
+        if (true) return 1.0F; // Nitrogen - Disable dynamic attack strength
         return Mth.clamp((this.attackStrengthTicker + a) / this.getCurrentItemAttackStrengthDelay(), 0.0F, 1.0F);
     }
 
