From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Biquaternions <biquaternions@serlith.net>
Date: Tue, 18 Aug 2026 22:24:24 -0500
Subject: [PATCH] Restore Legacy Health Regeneration


diff --git a/net/minecraft/world/food/FoodData.java b/net/minecraft/world/food/FoodData.java
index 38ccab90d09c92e6b70d1457931b64fb4e6ff7ec..daab0b55ebd09110eb1d172c8df09258b9780f4a 100644
--- a/net/minecraft/world/food/FoodData.java
+++ b/net/minecraft/world/food/FoodData.java
@@ -66,7 +66,8 @@ public class FoodData {
         }
 
         boolean naturalRegen = level.getGameRules().get(GameRules.NATURAL_HEALTH_REGENERATION);
-        if (naturalRegen && this.saturationLevel > 0.0F && player.isHurt() && this.foodLevel >= FoodConstants.MAX_FOOD) {
+        final me.biquaternions.nitrogen.entity.CombatProfile profile = player.getBukkitEntity().getCombatProfile(); // Nitrogen
+        if (profile.canRegenerateOnSaturation() && naturalRegen && this.saturationLevel > 0.0F && player.isHurt() && this.foodLevel >= FoodConstants.MAX_FOOD) { // Nitrogen - Toggle legacy regeneration
             this.tickTimer++;
             if (this.tickTimer >= this.saturatedRegenRate) { // CraftBukkit
                 float saturationSpent = Math.min(this.saturationLevel, 6.0F);
@@ -80,7 +81,7 @@ public class FoodData {
             if (this.tickTimer >= this.unsaturatedRegenRate) { // CraftBukkit - add regen rate manipulation
                 player.heal(1.0F, org.bukkit.event.entity.EntityRegainHealthEvent.RegainReason.SATIATED); // CraftBukkit - added RegainReason
                 // this.addExhaustion(6.0F); // CraftBukkit - EntityExhaustionEvent
-                player.causeFoodExhaustion(player.level().spigotConfig.regenExhaustion, org.bukkit.event.entity.EntityExhaustionEvent.ExhaustionReason.REGEN); // CraftBukkit - EntityExhaustionEvent // Spigot - Change to use configurable value
+                player.causeFoodExhaustion(profile.getRegenerationExhaustion(), org.bukkit.event.entity.EntityExhaustionEvent.ExhaustionReason.REGEN); // CraftBukkit - EntityExhaustionEvent // Spigot - Change to use configurable value // Nitrogen - Food exhaustion
                 this.tickTimer = 0;
             }
         } else if (this.foodLevel <= 0) {
