From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Biquaternions <biquaternions@serlith.net>
Date: Tue, 11 Aug 2026 14:19:36 -0500
Subject: [PATCH] Nitrogen Rebrand


diff --git a/src/main/java/gg/pufferfish/pufferfish/PufferfishVersionFetcher.java b/src/main/java/gg/pufferfish/pufferfish/PufferfishVersionFetcher.java
index 924e968f46507b5c44ac052f027f25eea3b9da08..b0266436e88052472cec07ea2f169b8fc9f87253 100644
--- a/src/main/java/gg/pufferfish/pufferfish/PufferfishVersionFetcher.java
+++ b/src/main/java/gg/pufferfish/pufferfish/PufferfishVersionFetcher.java
@@ -34,12 +34,12 @@ import static net.minecraft.server.MinecraftServer.COMPONENT_LOGGER;
 
 public class PufferfishVersionFetcher extends PaperVersionFetcher {
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(PufferfishVersionFetcher.class.getSimpleName());
+    private static final Logger LOGGER = LoggerFactory.getLogger("NitrogenVersionFetcher"); // Nitrogen - Rebrand
     private static final long CACHE_TIME = TimeUnit.MINUTES.toMillis(30);
 
-    private static final String REPOSITORY = "SerlithNetwork/Puffernot";
-    private static final String DOWNLOAD_PAGE = "https://github.com/SerlithNetwork/Puffernot/releases";
-    private static final TextColor COLOR_SECONDARY = TextColor.color(0x87, 0xDD, 0xFF);
+    private static final String REPOSITORY = "Biquaternions/Nitrogen"; // Nitrogen - Rebrand
+    private static final String DOWNLOAD_PAGE = "https://github.com/Biquaternions/Nitrogen/releases"; // Nitrogen - Rebrand
+    private static final TextColor COLOR_SECONDARY = TextColor.color(0xFF, 0xAB, 0xAB); // Nitrogen - Rebrand
 
     private static final int DISTANCE_ERROR = -1;
     private static final int DISTANCE_UNKNOWN = -2;
@@ -105,7 +105,7 @@ public class PufferfishVersionFetcher extends PaperVersionFetcher {
 
     private static int fetchDistanceFromSiteApi(final int actionsBuild) {
         try {
-            final URL buildsUrl = URI.create("https://version.serlith.net/api/v1/project/fetch/pufferfish/" + PufferfishVersionFetcher.BUILD_INFO.minecraftVersionId() + "/latest/build").toURL();
+            final URL buildsUrl = URI.create("https://version.serlith.net/api/v1/project/fetch/nitrogen/" + PufferfishVersionFetcher.BUILD_INFO.minecraftVersionId() + "/latest/build").toURL(); // Nitrogen - Rebrand
             final HttpURLConnection connection = (HttpURLConnection) buildsUrl.openConnection();
             connection.setConnectTimeout(5000);
             connection.setReadTimeout(5000);
@@ -116,7 +116,7 @@ public class PufferfishVersionFetcher extends PaperVersionFetcher {
                 final int latest = json.getAsInt();
                 return Math.max(latest - actionsBuild, 0);
             } catch (final JsonSyntaxException | MalformedJsonException ex) {
-                LOGGER.error("Error parsing json from Pufferfish's downloads API", ex);
+                LOGGER.error("Error parsing json from Nitrogen's downloads API", ex); // Nitrogen - Rebrand
                 return DISTANCE_ERROR;
             } catch (final SocketTimeoutException ex) {
                 return 0;
diff --git a/src/main/java/gg/pufferfish/pufferfish/compat/ServerConfigurations.java b/src/main/java/gg/pufferfish/pufferfish/compat/ServerConfigurations.java
index 5ad53edffc891bf1fc35977e7c09cfaaa72af4d1..b20d10be8eab3e2fb7190df5b5ac575143c0d8ca 100644
--- a/src/main/java/gg/pufferfish/pufferfish/compat/ServerConfigurations.java
+++ b/src/main/java/gg/pufferfish/pufferfish/compat/ServerConfigurations.java
@@ -25,7 +25,8 @@ public class ServerConfigurations {
         "spigot.yml",
         "config/paper-global.yml",
         "config/paper-world-defaults.yml",
-        "pufferfish.yml"
+        "pufferfish.yml",
+        "nitrogen.yml"
     };
 
     private static final String[] hiddenConfigEntries = new String[]{
diff --git a/src/main/java/gg/pufferfish/pufferfish/flare/PluginLookup.java b/src/main/java/gg/pufferfish/pufferfish/flare/PluginLookup.java
index 3acffffcb5c63ca0a63691c52808369e85494eed..92206e47ebd20fc9f95a2b0698b385692e1d1b01 100644
--- a/src/main/java/gg/pufferfish/pufferfish/flare/PluginLookup.java
+++ b/src/main/java/gg/pufferfish/pufferfish/flare/PluginLookup.java
@@ -33,6 +33,7 @@ public class PluginLookup {
             || name.startsWith("ca.spottedleaf.") || name.startsWith("com.sun.") || name.startsWith("org.jline.") || name.startsWith("org.bukkit.")
             || name.startsWith("org.spigotmc.") || name.startsWith("com.destroystokyo.paper") || name.startsWith("co.aikar.")
             || name.startsWith("io.netty.") || name.startsWith("com.velocitypowered.")
+            || name.startsWith("me.biquaternions.nitrogen.") // Nitrogen - Rebrand
         ) {
             return Optional.empty();
         }
