From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Biquaternions <biquaternions@serlith.net>
Date: Wed, 18 Jun 2025 18:00:05 -0500
Subject: [PATCH] Nitrogen Rebrand


diff --git a/src/main/java/io/papermc/paper/ServerBuildInfo.java b/src/main/java/io/papermc/paper/ServerBuildInfo.java
index b73e8d6aba63213127dfeece1f3eb331c9782d5d..0197ac67d21271243a3f17622aaab5a5ddbaa147 100644
--- a/src/main/java/io/papermc/paper/ServerBuildInfo.java
+++ b/src/main/java/io/papermc/paper/ServerBuildInfo.java
@@ -17,6 +17,7 @@ public interface ServerBuildInfo {
      */
     Key BRAND_PAPER_ID = Key.key("papermc", "paper");
     Key BRAND_PUFFERFISH_ID = Key.key("pufferfish", "pufferfish"); // Pufferfish - Rebrand
+    Key BRAND_NITROGEN_ID = Key.key("biquaternions", "nitrogen"); // Nitrogen - Rebrand
 
     /**
      * Gets the {@code ServerBuildInfo}.
