From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Biquaternions <biquaternions@serlith.net>
Date: Mon, 17 Aug 2026 22:07:18 -0500
Subject: [PATCH] Combat Profiles API


diff --git a/src/main/java/org/bukkit/entity/LivingEntity.java b/src/main/java/org/bukkit/entity/LivingEntity.java
index ab18c028fd800a7b2da91e5c32a4547beea1d92b..bf1efdbc2c860d88cfff69d90c747df02c3439c1 100644
--- a/src/main/java/org/bukkit/entity/LivingEntity.java
+++ b/src/main/java/org/bukkit/entity/LivingEntity.java
@@ -1559,4 +1559,23 @@ public interface LivingEntity extends Attributable, Damageable, ProjectileSource
      */
     @Nullable
     Color getWaypointColor();
+
+    // Nitrogen start - Combat Profiles API
+    /**
+     * Gets the entity combat profile
+     *
+     * @return combat profile
+     */
+    default me.biquaternions.nitrogen.entity.@org.jspecify.annotations.NonNull CombatProfile getCombatProfile() {
+        return null;
+    }
+
+    /**
+     * Sets the entity combat profile
+     *
+     * @param profile combat profile
+     */
+    default void setCombatProfile(me.biquaternions.nitrogen.entity.@org.jspecify.annotations.NonNull CombatProfile profile) {
+    }
+    // Nitrogen end - Combat Profiles API
 }
