From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Biquaternions <biquaternions@serlith.net>
Date: Wed, 12 Aug 2026 17:16:39 -0500
Subject: [PATCH] Disable Dynamic Attack Strength


diff --git a/net/minecraft/world/entity/player/Player.java b/net/minecraft/world/entity/player/Player.java
index 99830f9500e63b1bae397738a7243c5312827d5b..cf3e1d4f34c33f8d22a4d79de78041e2902a2ed5 100644
--- a/net/minecraft/world/entity/player/Player.java
+++ b/net/minecraft/world/entity/player/Player.java
@@ -1124,6 +1124,7 @@ public abstract class Player extends Avatar implements ContainerUser {
     }
 
     private boolean isSweepAttack(final boolean fullStrengthAttack, final boolean criticalAttack, final boolean knockbackAttack) {
+        if (true) return false; // Nitrogen - Disable sweep attack
         if (fullStrengthAttack && !criticalAttack && !knockbackAttack && this.onGround()) {
             double approximateSpeedSq = this.getKnownMovement().horizontalDistanceSqr();
             double maxSpeedForSweepAttack = this.getSpeed() * 2.5;
@@ -1963,6 +1964,7 @@ public abstract class Player extends Avatar implements ContainerUser {
     }
 
     public float getCurrentItemAttackStrengthDelay() {
+        if (true) return 0.0F; // Nitrogen - Disable dynamic attack strength - API has to reflect no delay
         return (float)(1.0 / this.getAttributeValue(Attributes.ATTACK_SPEED) * 20.0);
     }
 
@@ -1973,6 +1975,7 @@ public abstract class Player extends Avatar implements ContainerUser {
     }
 
     public float getAttackStrengthScale(final float a) {
+        if (true) return 1.0F; // Nitrogen - Disable dynamic attack strength
         return Mth.clamp((this.attackStrengthTicker + a) / this.getCurrentItemAttackStrengthDelay(), 0.0F, 1.0F);
     }
 
